docs/DEVELOPING.md

```diff
@@ -717,6 +717,188 @@ $ make ftest NAME=mssql
 
 ℹ️ The e2e test uses default values defined in [Configure Microsoft SQL connector](#configure-microsoft-sql-connector)
 
+
+## SharePoint Connector
+
+The [Elastic SharePoint connector](../connectors/sources/sharepoint.py) is provided in the Elastic connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+### Availability and prerequisites
+
+⚠️ _Currently, this connector is available in **beta** starting in 8.8_.
+Features in beta are subject to change and are not covered by the service level agreement (SLA) of features that have reached general availability (GA).
+
+Elastic versions 8.6.0+ are compatible with Elastic connector frameworks. Your deployment must include the Elasticsearch, Kibana, and Enterprise Search services.
+
+### Compatibility
+
+  Sharepoint Connector supports below Sharepoint Server versions:
+  - Sharepoint 2013
+  - Sharepoint 2016
+  - Sharepoint 2019
+
+
+### Create an Administrator Account For SharePoint Online:
+  - To create an administrator account, join the Microsoft 365 Developer Program and get the Microsoft 365 developer subscription with a new administrator account for 90 days. Use this administrator account for signing into SharePoint Online and Azure AD to fetch more details.
+  - Register your app with Microsoft Azure AD
+  - To use the Microsoft identity platform endpoint, you must register your app using the Azure app registration portal. You can use either a Microsoft account or a work or school account to register an app.
+  - To configure an app to use the OAuth 2.0 authorization code grant flow, you'll need to save the following values when registering the app:
+    - Client Id: `Application(client) ID` which you can find inside App registrations from azure portal
+    - Client Secret: It is a `Value` which you need to create  form `Certificates and secrets` from azure portal
+    - Tenant Id: Dedicated instance of azure AD  which you can find inside App registrations from azure portal
+    - Tenant name: Name of domain
+
+
+### SharePoint Online permissions required to run the connector:
+
+   Refer to the following documentation for setting [SharePoint Permissions](https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs)
+
+   - To set `DisableCustomAppAuthentication` to false we need to connect to SharePoint using Windows PowerShell and then run set-spotenant -DisableCustomAppAuthentication $false
```

Comment:
From the [above link](https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs) this seems like the wrong approach?

```
For new tenants, apps using an ACS app-only access token is disabled by default. We recommend using the Azure AD app-only model which is modern and more secure. But you can change the behavior by running 'set-spotenant -DisableCustomAppAuthentication $false' (needs the latest SharePoint admin PowerShell).
```

Why are we not using the "Azure AD app-only model (which is modern and more secure)"?

Reply:
In fact, shouldn't we just be using the client credentials flow? https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow#get-a-token

Reply:
@seanstory I tried using `Azure AD app-only model` but sharepoint is returning to me this page every time:
![Screenshot (95)](https://user-images.githubusercontent.com/125074214/235904718-40306b60-db2a-47a8-be4a-2e35a5e49180.png)